### ISSUE TITLE
Asp.Net Core ApplicationPart usage for plugin architecture

### ISSUE THREAD

**Issue by fsefacan** [nopSolutions/nopCommerce#2951]
nopCommerce version:  4.00

Steps to reproduce the problem: Plugin architecture >> ApplicationPart Usage

Hi guys,
I recommend you to try ApplicationPartManager, a new feature of dotnet core for plugin architecture.

Sample usage
```
//add module features support
builder.AddModuleFeatures(environment);
```

```
private static IMvcBuilder AddModuleFeatures(this IMvcBuilder builder, IHostingEnvironment environment)
{
    var fileNames = Directory.GetFiles("Modules", "*.dll", SearchOption.AllDirectories);

   foreach (string fileName in fileNames)
         builder.AddApplicationPart(
            AssemblyLoadContext.Default.LoadFromAssemblyPath(
                   Path.Combine(environment.ContentRootPath, fileName)));

    return builder;
 }
```

more information
https://github.com/aspnet/Mvc/issues/4572#issuecomment-216351033
https://github.com/aspnetboilerplate/aspnetboilerplate/blob/dev/src/Abp.AspNetCore/AspNetCore/AbpAspNetCoreModule.cs
https://www.codeproject.com/Articles/1109475/Modular-Web-Application-with-ASP-NET-Core

**Comment by AndreiMaz** [comment]
@skoshelev Sergey, please check

**Comment by skoshelev** [comment]
@fsefacan We do the same as this function does, please see more details under the following links:
https://github.com/aspnet/Mvc/blob/747420e5aa7cc2c7834cfb9731510286ded6fc03/src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/MvcCoreMvcCoreBuilderExtensions.cs#L141

https://github.com/nopSolutions/nopCommerce/blob/develop/src/Libraries/Nop.Core/Plugins/PluginManager.cs#L272

Closed #2951
